homeassistant/components/voip/assist_satellite.py

```diff
@@ -0,0 +1,165 @@
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .devices import VoIPDevice
+from .entity import VoIPEntity
+
+if TYPE_CHECKING:
+    from . import DomainData
+
+_LOGGER = logging.getLogger(__name__)
+
+_CONVERSATION_TIMEOUT_SEC: Final = 5 * 60  # 5 minutes
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up VoIP Assist satellite entity."""
+    domain_data: DomainData = hass.data[DOMAIN]
+
+    @callback
+    def async_add_device(device: VoIPDevice) -> None:
+        """Add device."""
+        async_add_entities([VoipAssistSatellite(hass, device)])
+
+    domain_data.devices.async_add_new_device_listener(async_add_device)
+
+    entities: list[VoIPEntity] = [
+        VoipAssistSatellite(hass, device) for device in domain_data.devices
+    ]
+
+    async_add_entities(entities)
+
+
+class VoipAssistSatellite(VoIPEntity, AssistSatelliteEntity):
+    """Assist satellite for VoIP devices."""
+
+    def __init__(self, hass: HomeAssistant, voip_device: VoIPDevice) -> None:
+        """Initialize an Assist satellite."""
+        VoIPEntity.__init__(self, voip_device)
+        AssistSatelliteEntity.__init__(self)
+
+        # Conversation id is reset after 5 minutes of inactivity
+        self._conversation_id: str | None = None
+        self._conversation_id_time: float | None = None
+
+        self._satellite_state: AssistSatelliteState = AssistSatelliteState.IDLE
+
+    @property
+    def state(self) -> AssistSatelliteState:
+        """Return the current state of the satellite."""
+        return self._satellite_state
+
+    @property
+    def supported_features(self) -> AssistSatelliteEntityFeature:
+        """Return supported features of the satellite."""
+        return (
+            AssistSatelliteEntityFeature.AUDIO_INPUT
+            | AssistSatelliteEntityFeature.AUDIO_OUTPUT
+        )
+
+    async def _async_accept_pipeline_from_satellite(
+        self,
+        context: Context,
+        event_callback: PipelineEventCallback,
+        audio_stream: AsyncIterable[bytes],
+        start_stage: PipelineStage = PipelineStage.STT,
+        end_stage: PipelineStage = PipelineStage.TTS,
+        wake_word_phrase: str | None = None,
+    ) -> None:
+        """Triggers an Assist pipeline in Home Assistant from a satellite."""
+
+        # Reset conversation id, if necessary
+        if (self._conversation_id_time is None) or (
+            (time.monotonic() - self._conversation_id_time) > _CONVERSATION_TIMEOUT_SEC
+        ):
+            self._conversation_id = None
+
+        if self._conversation_id is None:
+            self._conversation_id = str(uuid4())
+
+        # Update timeout
+        self._conversation_id_time = time.monotonic()
+
+        # Set entity state based on pipeline events
+        has_tts = False
+        self._set_state(AssistSatelliteState.LISTENING)
```

Comment:
This whole function needs to be defined inside the assist satellite base class and called by the VoIP satellite entity.

Reply:
Moved into the base class